relay-common/src/glob3.rs

```diff
@@ -99,6 +99,33 @@ impl<'de> Deserialize<'de> for GlobPatterns {
     }
 }
 
+/// Deserializes the [`GlobPatterns`] to a singular string.
+///
+/// This assumes there's only one pattern. Use in conjunction with [`deserialize_singular_pattern`].
+pub fn serialize_singular_pattern<S>(name: &GlobPatterns, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
I would much prefer if that was a newtype (or just straight up a separate type) instead which can only ever hold one pattern, `GlobPattern`?

Reply:
I was considering a separate type, but thought it'd add too much code repetition. But I like the newtype idea. Maybe something like this?

```Rust
/// A singular glob-pattern.
pub struct GlobPattern(GlobPatterns);

impl GlobPattern {
    ///
    pub fn new(pattern: String) -> Self {
        Self(GlobPatterns::new(vec![pattern]))
    }

    ///
    pub fn is_match<S>(&self, message: S) -> bool
    where
        S: AsRef<[u8]>,
    {
        self.0.is_match(message)
    }
}
```

Reply:
ok i went ahead and did it, since it was so simple anyway